// @webstudio-is/sdk-components-react-radix/lib/accordion.js
import { jsx as d } from "react/jsx-runtime";
import { forwardRef as m } from "react";
import { Root as f, Item as g, Header as p, Trigger as A, Content as l } from "@radix-ui/react-accordion";
import { getIndexWithinAncestorFromComponentProps as u, getClosestInstance as a } from "@webstudio-is/react-sdk/runtime";
const x = m((n, o) => /* @__PURE__ */ d(f, { ref: o, type: "single", ...n })), $ = m(({ value: n, ...o }, e) => {
  const t = u(o);
  return /* @__PURE__ */ d(g, { ref: e, value: n ?? t ?? "", ...o });
}), k = p, v = A, w = l, r = "@webstudio-is/sdk-components-react-radix", y = {
  onNavigatorSelect: (n, o) => {
    var e;
    for (const t of o.instancePath)
      if (t.component === `${r}:AccordionContent`) {
        const i = a(
          o.instancePath,
          t,
          `${r}:Accordion`
        ), c = a(
          o.instancePath,
          t,
          `${r}:AccordionItem`
        );
        if (i && c) {
          const s = n.getPropValue(c, "value") ?? ((e = n.indexesWithinAncestors.get(c.id)) == null ? void 0 : e.toString());
          s && n.setMemoryProp(i, "value", s);
        }
      }
  }
};
export {
  x as Accordion,
  w as AccordionContent,
  k as AccordionHeader,
  $ as AccordionItem,
  v as AccordionTrigger,
  y as hooksAccordion
};
