// @webstudio-is/sdk-components-react-radix/lib/collapsible.ws.js
import { CollapsibleIcon as t, TriggerIcon as o, ContentIcon as n } from "@webstudio-is/icons/svg";
import { div as l } from "@webstudio-is/sdk/normalize.css";
import { propsCollapsible as p, propsCollapsibleTrigger as i, propsCollapsibleContent as a } from "./__generated__/collapsible.props.js";
import { getButtonStyles as r } from "./theme/styles.js";
const e = {
  div: l
}, g = {
  category: "radix",
  order: 5,
  type: "container",
  presetStyle: e,
  icon: t,
  description: "An interactive component which expands and collapses some content, triggered by a button.",
  template: [
    {
      type: "instance",
      component: "Collapsible",
      props: [],
      children: [
        {
          type: "instance",
          component: "CollapsibleTrigger",
          children: [
            {
              type: "instance",
              component: "Button",
              styles: r("outline"),
              children: [
                {
                  type: "text",
                  value: "Click to toggle content",
                  placeholder: !0
                }
              ]
            }
          ]
        },
        {
          type: "instance",
          component: "CollapsibleContent",
          children: [
            {
              type: "instance",
              component: "Text",
              children: [
                {
                  type: "text",
                  value: "Collapsible Content",
                  placeholder: !0
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}, b = {
  category: "hidden",
  type: "container",
  icon: o,
  stylable: !1,
  detachable: !1
}, m = {
  category: "hidden",
  type: "container",
  presetStyle: e,
  icon: n,
  detachable: !1
}, y = {
  props: {
    ...p,
    onOpenChange: {
      type: "action",
      control: "action",
      required: !1
    }
  },
  initialProps: ["open", "onOpenChange"]
}, h = {
  props: i
}, u = {
  props: a
};
export {
  g as metaCollapsible,
  m as metaCollapsibleContent,
  b as metaCollapsibleTrigger,
  y as propsMetaCollapsible,
  u as propsMetaCollapsibleContent,
  h as propsMetaCollapsibleTrigger
};
