// @webstudio-is/sdk-components-react-radix/lib/label.ws.js
import { LabelIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as o } from "@webstudio-is/react-sdk";
import { label as r } from "@webstudio-is/sdk/normalize.css";
import { props as t } from "./__generated__/label.props.js";
import { text as a, font as l, leading as p } from "./theme/tailwind-classes.js";
const n = {
  label: r
}, d = {
  category: "radix",
  order: 102,
  type: "container",
  icon: e,
  presetStyle: n,
  states: o,
  description: "An accessible label to describe the purpose of an input. Match the “For” property on the label with the “ID” of the input to connect them.",
  template: [
    {
      type: "instance",
      component: "Label",
      styles: [
        // text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70
        a("sm"),
        l("medium"),
        p("none")
        // We are not supporting peer like styles yet
      ].flat(),
      children: [{ type: "text", value: "Form Label", placeholder: !0 }]
    }
  ]
}, f = {
  props: {
    ...t,
    htmlFor: {
      ...t.htmlFor,
      label: "For"
    }
  },
  initialProps: ["id", "className", "htmlFor"]
};
export {
  d as meta,
  f as propsMeta
};
