// @webstudio-is/sdk-components-react-radix/lib/navigation-menu.js
import { jsx as r } from "react/jsx-runtime";
import * as a from "@radix-ui/react-navigation-menu";
import { ReactSdkContext as v, getIndexWithinAncestorFromComponentProps as l, getClosestInstance as d } from "@webstudio-is/react-sdk/runtime";
import { forwardRef as c, useContext as f, Children as m } from "react";
const p = c(({ value: o, ...n }, e) => {
  const { renderer: t } = f(v);
  let i = o;
  return t === "canvas" && (i = i === "" ? "-1" : i), /* @__PURE__ */ r(a.Root, { ref: e, value: i, ...n });
}), C = a.List, P = a.Viewport, k = a.Content, x = c(({ value: o, ...n }, e) => {
  const t = l(n);
  return /* @__PURE__ */ r(a.Item, { ref: e, value: o ?? t, ...n });
}), A = c(({ children: o, ...n }, e) => {
  const t = m.toArray(o)[0];
  return /* @__PURE__ */ r(a.Link, { asChild: !0, ref: e, ...n, children: t ?? /* @__PURE__ */ r("a", { children: "Add link component" }) });
}), I = c(({ children: o, ...n }, e) => {
  const t = m.toArray(o)[0];
  return /* @__PURE__ */ r(a.Trigger, { asChild: !0, ref: e, ...n, children: t ?? /* @__PURE__ */ r("button", { children: "Add button or link" }) });
}), s = "@webstudio-is/sdk-components-react-radix", $ = {
  onNavigatorUnselect: (o, n) => {
    for (const e of n.instancePath)
      if (e.component === `${s}:NavigationMenuContent`) {
        const t = d(
          n.instancePath,
          e,
          `${s}:NavigationMenu`
        );
        t && o.setMemoryProp(t, "value", void 0);
      }
  },
  onNavigatorSelect: (o, n) => {
    var e;
    for (const t of n.instancePath)
      if (t.component === `${s}:NavigationMenuContent`) {
        const i = d(
          n.instancePath,
          t,
          `${s}:NavigationMenu`
        ), u = d(
          n.instancePath,
          t,
          `${s}:NavigationMenuItem`
        );
        if (u === void 0 || i === void 0)
          return;
        const g = o.getPropValue(u, "value") ?? ((e = o.indexesWithinAncestors.get(u.id)) == null ? void 0 : e.toString());
        g && o.setMemoryProp(i, "value", g);
      }
  }
};
export {
  p as NavigationMenu,
  k as NavigationMenuContent,
  x as NavigationMenuItem,
  A as NavigationMenuLink,
  C as NavigationMenuList,
  I as NavigationMenuTrigger,
  P as NavigationMenuViewport,
  $ as hooksNavigationMenu
};
