// @webstudio-is/sdk-components-react-radix/lib/popover.js
import { jsx as n } from "react/jsx-runtime";
import { forwardRef as c, Children as P } from "react";
import * as s from "@radix-ui/react-popover";
import { getClosestInstance as a } from "@webstudio-is/react-sdk/runtime";
const l = c((o, t) => /* @__PURE__ */ n(s.Root, { ...o })), v = c(({ children: o, ...t }, e) => {
  const r = P.toArray(o)[0];
  return /* @__PURE__ */ n(s.Trigger, { asChild: !0, ref: e, ...t, children: r ?? /* @__PURE__ */ n("button", { children: "Add button or link" }) });
}), h = c(
  ({ sideOffset: o = 4, align: t = "center", hideWhenDetached: e = !0, ...r }, p) => /* @__PURE__ */ n(s.Portal, { children: /* @__PURE__ */ n(
    s.Content,
    {
      ref: p,
      align: "center",
      sideOffset: o,
      hideWhenDetached: e,
      ...r
    }
  ) })
), i = "@webstudio-is/sdk-components-react-radix", g = {
  onNavigatorUnselect: (o, t) => {
    for (const e of t.instancePath)
      if (e.component === `${i}:PopoverContent`) {
        const r = a(
          t.instancePath,
          e,
          `${i}:Popover`
        );
        r && o.setMemoryProp(r, "open", void 0);
      }
  },
  onNavigatorSelect: (o, t) => {
    for (const e of t.instancePath)
      if (e.component === `${i}:PopoverContent`) {
        const r = a(
          t.instancePath,
          e,
          `${i}:Popover`
        );
        r && o.setMemoryProp(r, "open", !0);
      }
  }
};
export {
  l as Popover,
  h as PopoverContent,
  v as PopoverTrigger,
  g as hooksPopover
};
