// @webstudio-is/sdk-components-react-radix/lib/props.js
import { propsMetaCollapsible as a, propsMetaCollapsibleContent as t, propsMetaCollapsibleTrigger as r } from "./collapsible.ws.js";
import { propsMetaDialog as s, propsMetaDialogClose as i, propsMetaDialogContent as n, propsMetaDialogDescription as l, propsMetaDialogOverlay as M, propsMetaDialogTitle as g, propsMetaDialogTrigger as c } from "./dialog.ws.js";
import { propsMetaPopover as C, propsMetaPopoverContent as m, propsMetaPopoverTrigger as d } from "./popover.ws.js";
import { propsMetaTooltip as v, propsMetaTooltipContent as b, propsMetaTooltipTrigger as S } from "./tooltip.ws.js";
import { propsMetaTabs as I, propsMetaTabsContent as D, propsMetaTabsList as N, propsMetaTabsTrigger as f } from "./tabs.ws.js";
import { propsMeta as A } from "./label.ws.js";
import { propsMetaAccordion as L, propsMetaAccordionContent as k, propsMetaAccordionHeader as G, propsMetaAccordionItem as P, propsMetaAccordionTrigger as R } from "./accordion.ws.js";
import { propsMetaNavigationMenu as y, propsMetaNavigationMenuContent as H, propsMetaNavigationMenuItem as O, propsMetaNavigationMenuLink as j, propsMetaNavigationMenuList as q, propsMetaNavigationMenuTrigger as z, propsMetaNavigationMenuViewport as B } from "./navigation-menu.ws.js";
import { propsMetaSelect as F, propsMetaSelectContent as J, propsMetaSelectItem as K, propsMetaSelectItemIndicator as Q, propsMetaSelectItemText as U, propsMetaSelectTrigger as W, propsMetaSelectValue as X, propsMetaSelectViewport as Y } from "./select.ws.js";
import { propsMetaSwitch as _, propsMetaSwitchThumb as $ } from "./switch.ws.js";
import { propsMetaCheckbox as eo, propsMetaCheckboxIndicator as ao } from "./checkbox.ws.js";
import { propsMetaRadioGroup as ro, propsMetaRadioGroupIndicator as po, propsMetaRadioGroupItem as so } from "./radio-group.ws.js";
export {
  L as Accordion,
  k as AccordionContent,
  G as AccordionHeader,
  P as AccordionItem,
  R as AccordionTrigger,
  eo as Checkbox,
  ao as CheckboxIndicator,
  a as Collapsible,
  t as CollapsibleContent,
  r as CollapsibleTrigger,
  s as Dialog,
  i as DialogClose,
  n as DialogContent,
  l as DialogDescription,
  M as DialogOverlay,
  g as DialogTitle,
  c as DialogTrigger,
  A as Label,
  y as NavigationMenu,
  H as NavigationMenuContent,
  O as NavigationMenuItem,
  j as NavigationMenuLink,
  q as NavigationMenuList,
  z as NavigationMenuTrigger,
  B as NavigationMenuViewport,
  C as Popover,
  m as PopoverContent,
  d as PopoverTrigger,
  ro as RadioGroup,
  po as RadioGroupIndicator,
  so as RadioGroupItem,
  F as Select,
  J as SelectContent,
  K as SelectItem,
  Q as SelectItemIndicator,
  U as SelectItemText,
  W as SelectTrigger,
  X as SelectValue,
  Y as SelectViewport,
  _ as Switch,
  $ as SwitchThumb,
  I as Tabs,
  D as TabsContent,
  N as TabsList,
  f as TabsTrigger,
  v as Tooltip,
  b as TooltipContent,
  S as TooltipTrigger
};
