// @webstudio-is/sdk-components-react-radix/lib/select.js
import { jsx as c } from "react/jsx-runtime";
import { forwardRef as s, useContext as l } from "react";
import { Root as m, Trigger as p, Value as f, Portal as S, Content as d, Viewport as u, Item as I, ItemIndicator as P, ItemText as g } from "@radix-ui/react-select";
import { ReactSdkContext as x, getClosestInstance as i } from "@webstudio-is/react-sdk/runtime";
const v = s(
  ({ value: e, defaultValue: t, ...o }, n) => /* @__PURE__ */ c(m, { ...o, defaultValue: e ?? t })
), T = s((e, t) => {
  const { renderer: o } = l(x);
  return /* @__PURE__ */ c(p, { onPointerDown: o === "canvas" ? (a) => {
    a.preventDefault();
  } : void 0, ref: t, ...e });
}), $ = s((e, t) => /* @__PURE__ */ c(f, { ref: t, ...e })), k = s((e, t) => /* @__PURE__ */ c(S, { children: /* @__PURE__ */ c(d, { ref: t, ...e, position: "popper" }) })), D = u, R = I, y = P, M = g, r = "@webstudio-is/sdk-components-react-radix", N = {
  onNavigatorUnselect: (e, t) => {
    for (const o of t.instancePath)
      if (o.component === `${r}:SelectContent`) {
        const n = i(
          t.instancePath,
          o,
          `${r}:Select`
        );
        n && e.setMemoryProp(n, "open", void 0);
      }
  },
  onNavigatorSelect: (e, t) => {
    for (const o of t.instancePath)
      if (o.component === `${r}:SelectContent`) {
        const n = i(
          t.instancePath,
          o,
          `${r}:Select`
        );
        n && e.setMemoryProp(n, "open", !0);
      }
  }
};
export {
  v as Select,
  k as SelectContent,
  R as SelectItem,
  y as SelectItemIndicator,
  M as SelectItemText,
  T as SelectTrigger,
  $ as SelectValue,
  D as SelectViewport,
  N as hooksSelect
};
