// @webstudio-is/sdk-components-react-radix/lib/sheet.js
import { jsx as s } from "react/jsx-runtime";
import { forwardRef as f } from "react";
import { getClosestInstance as r } from "@webstudio-is/react-sdk/runtime";
import { DialogContent as h } from "./dialog.js";
f(
  ({ tag: n = "nav", side: t = "left", role: e = "navigation", children: o, ...i }, c) => /* @__PURE__ */ s(
    h,
    {
      asChild: !0,
      "data-side": t,
      role: e,
      ...i,
      children: /* @__PURE__ */ s(n, { ref: c, children: o })
    }
  )
);
const a = "@webstudio-is/sdk-components-react-radix", P = {
  onNavigatorUnselect: (n, t) => {
    for (const e of t.instancePath)
      if (e.component === `${a}:SheetOverlay`) {
        const o = r(
          t.instancePath,
          e,
          `${a}:Sheet`
        );
        o && n.setMemoryProp(o, "open", void 0);
      }
  },
  onNavigatorSelect: (n, t) => {
    for (const e of t.instancePath)
      if (e.component === `${a}:SheetOverlay`) {
        const o = r(
          t.instancePath,
          e,
          `${a}:Sheet`
        );
        o && n.setMemoryProp(o, "open", !0);
      }
  }
};
export {
  P as hooksSheet
};
