// @webstudio-is/sdk-components-react-radix/lib/tabs.js
import { jsx as i } from "react/jsx-runtime";
import { forwardRef as c } from "react";
import { Root as f, List as g, Trigger as b, Content as d } from "@radix-ui/react-tabs";
import { getIndexWithinAncestorFromComponentProps as m, getClosestInstance as u } from "@webstudio-is/react-sdk/runtime";
const x = f, C = g, P = c(({ value: n, ...t }, e) => {
  const o = m(t);
  return /* @__PURE__ */ i(b, { ref: e, value: n ?? o ?? "", ...t });
}), v = c(({ value: n, ...t }, e) => {
  const o = m(t);
  return /* @__PURE__ */ i(d, { ref: e, value: n ?? o ?? "", ...t });
}), a = "@webstudio-is/sdk-components-react-radix", k = {
  onNavigatorSelect: (n, t) => {
    var e;
    for (const o of t.instancePath)
      if (o.component === `${a}:TabsContent`) {
        const s = u(
          t.instancePath,
          o,
          `${a}:Tabs`
        ), r = n.getPropValue(o, "value") ?? ((e = n.indexesWithinAncestors.get(o.id)) == null ? void 0 : e.toString());
        s && r && n.setMemoryProp(s, "value", r);
      }
  }
};
export {
  x as Tabs,
  v as TabsContent,
  C as TabsList,
  P as TabsTrigger,
  k as hooksTabs
};
