// @webstudio-is/sdk/lib/__generated__/normalize.css.js
// src/__generated__/normalize.css.ts
var div = [
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "outlineWidth", value: { type: "unit", unit: "px", value: 1 } }
];
var address = div;
var article = div;
var aside = div;
var figure = div;
var footer = div;
var header = div;
var main = div;
var nav = div;
var section = div;
var form = div;
var label = div;
var time = div;
var h1 = div;
var h2 = div;
var h3 = div;
var h4 = div;
var h5 = div;
var h6 = div;
var i = div;
var img = div;
var a = div;
var li = div;
var ul = div;
var ol = div;
var p = div;
var span = div;
var html = [
  { property: "display", value: { type: "keyword", value: "grid" } },
  { property: "minHeight", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "fontFamily",
    value: { type: "fontFamily", value: ["Arial", "Roboto", "sans-serif"] }
  },
  { property: "fontSize", value: { type: "unit", unit: "px", value: 16 } },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.2 }
  },
  {
    property: "whiteSpaceCollapse",
    value: { type: "keyword", value: "preserve" }
  }
];
var body = [
  { property: "marginTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "marginRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "marginBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "marginLeft", value: { type: "unit", unit: "number", value: 0 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var hr = [
  { property: "height", value: { type: "unit", unit: "number", value: 0 } },
  { property: "color", value: { type: "keyword", value: "inherit" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var b = [
  {
    property: "fontWeight",
    value: { type: "unit", unit: "number", value: 700 }
  },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var strong = b;
var code = [
  {
    property: "fontFamily",
    value: {
      type: "fontFamily",
      value: [
        "ui-monospace",
        "SFMono-Regular",
        "Consolas",
        "Liberation Mono",
        "Menlo",
        "monospace"
      ]
    }
  },
  { property: "fontSize", value: { type: "unit", unit: "em", value: 1 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var kbd = code;
var samp = code;
var pre = code;
var small = [
  { property: "fontSize", value: { type: "unit", unit: "%", value: 80 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var sub = [
  { property: "fontSize", value: { type: "unit", unit: "%", value: 75 } },
  { property: "lineHeight", value: { type: "unit", unit: "number", value: 0 } },
  { property: "position", value: { type: "keyword", value: "relative" } },
  { property: "verticalAlign", value: { type: "keyword", value: "baseline" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "bottom", value: { type: "unit", unit: "em", value: -0.25 } }
];
var sup = [
  { property: "fontSize", value: { type: "unit", unit: "%", value: 75 } },
  { property: "lineHeight", value: { type: "unit", unit: "number", value: 0 } },
  { property: "position", value: { type: "keyword", value: "relative" } },
  { property: "verticalAlign", value: { type: "keyword", value: "baseline" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "top", value: { type: "unit", unit: "em", value: -0.5 } }
];
var table = [
  { property: "textIndent", value: { type: "unit", unit: "number", value: 0 } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "borderTopColor", value: { type: "keyword", value: "inherit" } },
  {
    property: "borderRightColor",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "borderBottomColor",
    value: { type: "keyword", value: "inherit" }
  },
  { property: "borderLeftColor", value: { type: "keyword", value: "inherit" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } }
];
var input = [
  { property: "fontFamily", value: { type: "keyword", value: "inherit" } },
  { property: "fontSize", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "marginTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "marginRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "marginBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "marginLeft", value: { type: "unit", unit: "number", value: 0 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "borderTopStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderRightStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderBottomStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderLeftStyle", value: { type: "keyword", value: "solid" } }
];
var textarea = input;
var optgroup = [
  { property: "fontFamily", value: { type: "keyword", value: "inherit" } },
  { property: "fontSize", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "marginTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "marginRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "marginBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "marginLeft", value: { type: "unit", unit: "number", value: 0 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var radio = [
  { property: "fontFamily", value: { type: "keyword", value: "inherit" } },
  { property: "fontSize", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "marginTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "marginRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "marginBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "marginLeft", value: { type: "unit", unit: "number", value: 0 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "borderTopStyle", value: { type: "keyword", value: "none" } },
  { property: "borderRightStyle", value: { type: "keyword", value: "none" } },
  { property: "borderBottomStyle", value: { type: "keyword", value: "none" } },
  { property: "borderLeftStyle", value: { type: "keyword", value: "none" } }
];
var checkbox = radio;
var button = [
  { property: "fontFamily", value: { type: "keyword", value: "inherit" } },
  { property: "fontSize", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "marginTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "marginRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "marginBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "marginLeft", value: { type: "unit", unit: "number", value: 0 } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "borderTopStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderRightStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderBottomStyle", value: { type: "keyword", value: "solid" } },
  { property: "borderLeftStyle", value: { type: "keyword", value: "solid" } },
  { property: "textTransform", value: { type: "keyword", value: "none" } }
];
var select = button;
var legend = [
  { property: "paddingTop", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "paddingRight",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "paddingBottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "paddingLeft",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var progress = [
  { property: "verticalAlign", value: { type: "keyword", value: "baseline" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var summary = [
  { property: "display", value: { type: "keyword", value: "list-item" } },
  { property: "boxSizing", value: { type: "keyword", value: "border-box" } },
  { property: "borderTopWidth", value: { type: "unit", unit: "px", value: 1 } },
  {
    property: "borderRightWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
export {
  a,
  address,
  article,
  aside,
  b,
  body,
  button,
  checkbox,
  code,
  div,
  figure,
  footer,
  form,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  header,
  hr,
  html,
  i,
  img,
  input,
  kbd,
  label,
  legend,
  li,
  main,
  nav,
  ol,
  optgroup,
  p,
  pre,
  progress,
  radio,
  samp,
  section,
  select,
  small,
  span,
  strong,
  sub,
  summary,
  sup,
  table,
  textarea,
  time,
  ul
};
