// @webstudio-is/sdk-components-react/lib/blockquote.ws.js
import { BlockquoteIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { props as p } from "./__generated__/blockquote.props.js";
const r = {
  blockquote: [
    {
      property: "marginTop",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "marginRight",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "marginBottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "marginLeft",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "paddingTop",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "paddingBottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "paddingLeft",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "paddingRight",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "borderLeftWidth",
      value: { type: "unit", value: 5, unit: "px" }
    },
    {
      property: "borderLeftStyle",
      value: { type: "keyword", value: "solid" }
    },
    {
      property: "borderLeftColor",
      value: { type: "rgb", r: 226, g: 226, b: 226, alpha: 1 }
    }
  ]
}, i = {
  category: "text",
  type: "container",
  description: "Use to style a quote from an external source like an article or book.",
  icon: e,
  states: t,
  presetStyle: r,
  order: 3,
  template: [
    {
      type: "instance",
      component: "Blockquote",
      children: [
        {
          type: "text",
          value: "Blockquote text you can edit",
          placeholder: !0
        }
      ]
    }
  ]
}, n = {
  props: p,
  initialProps: ["id", "className", "cite"]
};
export {
  i as meta,
  n as propsMeta
};
