// @webstudio-is/sdk-components-react/lib/button.ws.js
import { ButtonElementIcon as t } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/react-sdk";
import { button as o } from "@webstudio-is/sdk/normalize.css";
import { props as n } from "./__generated__/button.props.js";
const a = {
  button: o
}, p = {
  category: "forms",
  type: "container",
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  description: "Use a button to submit forms or trigger actions within a page. Do not use a button to navigate users to another resource or another page - that’s what a link is used for.",
  icon: t,
  presetStyle: a,
  states: [
    ...e,
    { selector: ":disabled", label: "Disabled" },
    { selector: ":enabled", label: "Enabled" }
  ],
  order: 2,
  template: [
    {
      type: "instance",
      component: "Button",
      children: [
        {
          type: "text",
          value: "Button text you can edit",
          placeholder: !0
        }
      ]
    }
  ]
}, c = {
  props: n,
  initialProps: ["id", "className", "type", "aria-label"]
};
export {
  p as meta,
  c as propsMeta
};
