// @webstudio-is/sdk-components-react/lib/checkbox.ws.js
import { CheckboxCheckedIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { checkbox as o } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/checkbox.props.js";
const a = {
  input: [
    ...o,
    {
      property: "marginRight",
      value: { type: "unit", unit: "em", value: 0.5 }
    }
  ]
}, s = {
  category: "forms",
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  type: "control",
  description: "Use within a form to allow your users to toggle between checked and not checked. Group checkboxes by matching their “Name” properties. Unlike radios, any number of checkboxes in a group can be checked.",
  icon: e,
  presetStyle: a,
  order: 6,
  states: [
    ...t,
    { selector: ":checked", label: "Checked" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" },
    { selector: ":disabled", label: "Disabled" },
    { selector: ":enabled", label: "Enabled" },
    { selector: ":read-only", label: "Read Only" },
    { selector: ":read-write", label: "Read Write" }
  ],
  template: [
    {
      type: "instance",
      component: "Label",
      label: "Checkbox Field",
      children: [
        { type: "instance", component: "Checkbox", children: [] },
        {
          type: "instance",
          component: "Text",
          label: "Checkbox Label",
          props: [{ type: "string", name: "tag", value: "span" }],
          children: [{ type: "text", value: "Checkbox", placeholder: !0 }]
        }
      ]
    }
  ]
}, p = {
  props: r,
  initialProps: ["id", "className", "name", "value", "required", "checked"]
};
export {
  s as meta,
  p as propsMeta
};
