// @webstudio-is/sdk-components-react/lib/code-text.ws.js
import { CodeTextIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { code as o } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/code-text.props.js";
const p = {
  code: [
    ...o,
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    },
    {
      property: "whiteSpaceCollapse",
      value: { type: "keyword", value: "preserve" }
    },
    {
      property: "textWrapMode",
      value: { type: "keyword", value: "wrap" }
    },
    {
      property: "paddingLeft",
      value: { type: "unit", value: 0.2, unit: "em" }
    },
    {
      property: "paddingRight",
      value: { type: "unit", value: 0.2, unit: "em" }
    },
    {
      property: "backgroundColor",
      value: { type: "rgb", r: 238, g: 238, b: 238, alpha: 1 }
    }
  ]
}, d = {
  category: "general",
  type: "embed",
  description: "Use this component when you want to display code as text on the page.",
  icon: e,
  constraints: {
    relation: "ancestor",
    component: { $neq: "CodeText" }
  },
  states: t,
  presetStyle: p,
  order: 9
}, s = {
  props: {
    ...r,
    code: {
      required: !0,
      control: "codetext",
      type: "string"
    }
  },
  initialProps: ["id", "className", "lang", "code"]
};
export {
  d as meta,
  s as propsMeta
};
