// @webstudio-is/sdk-components-react/lib/heading.ws.js
import { HeadingIcon as t } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/react-sdk";
import { h1 as n, h2 as o, h3 as a, h4 as r, h5 as i, h6 as s } from "@webstudio-is/sdk/normalize.css";
import { props as p } from "./__generated__/heading.props.js";
const c = {
  h1: n,
  h2: o,
  h3: a,
  h4: r,
  h5: i,
  h6: s
}, l = {
  category: "text",
  type: "container",
  description: "Use HTML headings to structure and organize content. Use the Tag property in settings to change the heading level (h1-h6).",
  icon: t,
  constraints: {
    relation: "ancestor",
    component: { $neq: "Heading" }
  },
  states: e,
  presetStyle: c,
  order: 1,
  template: [
    {
      type: "instance",
      component: "Heading",
      children: [
        {
          type: "text",
          value: "Heading text you can edit",
          placeholder: !0
        }
      ]
    }
  ]
}, y = {
  props: p,
  initialProps: ["id", "className", "tag"]
};
export {
  l as meta,
  y as propsMeta
};
