// @webstudio-is/sdk-components-react/lib/html-embed.ws.js
import { EmbedIcon as t } from "@webstudio-is/icons/svg";
import { props as e } from "./__generated__/html-embed.props.js";
const i = {
  div: [
    {
      property: "display",
      value: { type: "keyword", value: "contents" }
    },
    {
      property: "whiteSpaceCollapse",
      value: { type: "keyword", value: "collapse" }
    }
  ]
}, a = {
  category: "general",
  type: "embed",
  label: "HTML Embed",
  description: "Used to add HTML code to the page, such as an SVG or script.",
  icon: t,
  presetStyle: i,
  order: 8
}, r = {
  props: {
    ...e,
    clientOnly: {
      ...e.clientOnly,
      description: "Activate it for any scripts that can mutate the DOM or introduce interactivity. This only affects the published site."
    },
    executeScriptOnCanvas: {
      ...e.executeScriptOnCanvas,
      label: "Run scripts on canvas",
      description: "Dangerously allow script execution on canvas without switching to preview mode. This only affects build mode, but may result in unwanted side effects inside builder!"
    },
    code: {
      required: !0,
      control: "code",
      language: "html",
      type: "string"
    }
  },
  initialProps: ["className", "clientOnly", "executeScriptOnCanvas"]
};
export {
  a as meta,
  r as propsMeta
};
