// @webstudio-is/sdk-components-react/lib/image.js
import { jsx as u } from "react/jsx-runtime";
import { forwardRef as v, useContext as x } from "react";
import { Image as y } from "@webstudio-is/image";
import { ReactSdkContext as I } from "@webstudio-is/react-sdk/runtime";
const p = v(
  ({
    loading: s = "lazy",
    width: r,
    height: a,
    optimize: o = !0,
    decoding: d,
    // @todo: it's a hack made for the builder and should't be in the runtime at all.
    $webstudio$canvasOnly$assetId: f,
    ...t
  }, l) => {
    const e = String(t.src ?? ""), { imageLoader: N, renderer: g, assetBaseUrl: m } = x(I);
    let c = d, n = e;
    g === "canvas" && (s = "eager", c = "sync", n = f ?? e, r !== void 0 && a !== void 0 && Number.isNaN(r) && Number.isNaN(a) && (o = !1, r = void 0, a = void 0));
    let i = e;
    return e.startsWith(m) && (i = e.slice(m.length)), /* @__PURE__ */ u(
      y,
      {
        loading: s,
        decoding: c,
        optimize: o,
        width: r,
        height: a,
        ...t,
        loader: N,
        src: i,
        ref: l
      },
      n
    );
  }
);
p.displayName = "Image";
export {
  p as Image
};
