// @webstudio-is/sdk-components-react/lib/image.ws.js
import { ImageIcon as o } from "@webstudio-is/icons/svg";
import { defaultStates as r } from "@webstudio-is/react-sdk";
import { img as i } from "@webstudio-is/sdk/normalize.css";
import { props as e } from "./__generated__/image.props.js";
const a = {
  img: [
    ...i,
    // Otherwise on new image insert onto canvas it can overfit screen size multiple times
    {
      property: "maxWidth",
      value: { type: "unit", unit: "%", value: 100 }
    },
    // inline | inline-block is not suitable because without line-height: 0 on the parent you get unsuitable spaces/margins
    // see https://stackoverflow.com/questions/24771194/is-the-margin-of-inline-block-4px-is-static-for-all-browsers
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    },
    // Set image height to "auto" to reduce layout shift, improving compatibility across browsers like Safari.
    // Unlike "fit-content," "auto" preserves the aspect ratio when the width exceeds max-width. (in Safari)
    // See https://web.dev/articles/optimize-cls#best_practice_for_setting_image_dimensions
    {
      property: "height",
      value: { type: "keyword", value: "auto" }
    }
  ]
}, l = {
  category: "media",
  type: "embed",
  description: "Add an image asset to the page. Webstudio automatically converts images to WebP or AVIF format and makes them responsive for best performance.",
  icon: o,
  states: r,
  presetStyle: a,
  order: 0
}, p = {
  src: {
    type: "string",
    control: "file",
    label: "Source",
    required: !1
  }
}, t = e.optimize;
t.description = "Optimize the image for enhanced performance.";
const d = {
  props: {
    ...e,
    ...p,
    optimize: t
  },
  initialProps: [
    "id",
    "className",
    "src",
    "width",
    "height",
    "alt",
    "loading",
    "optimize"
  ]
};
export {
  l as meta,
  d as propsMeta,
  p as propsOverrides
};
