// @webstudio-is/sdk-components-react/lib/input.ws.js
import { FormTextFieldIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { input as o } from "@webstudio-is/sdk/normalize.css";
import { props as l } from "./__generated__/input.props.js";
const r = {
  input: [
    ...o,
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    }
  ]
}, p = {
  category: "forms",
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  type: "control",
  label: "Text Input",
  description: "A single-line text input for collecting string data from your users.",
  icon: e,
  presetStyle: r,
  order: 3,
  states: [
    ...t,
    { selector: "::placeholder", label: "Placeholder" },
    { selector: ":valid", label: "Valid" },
    { selector: ":invalid", label: "Invalid" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ]
}, c = {
  props: l,
  initialProps: [
    "id",
    "className",
    "name",
    "value",
    "type",
    "placeholder",
    "required",
    "autoFocus"
  ]
};
export {
  p as meta,
  c as propsMeta
};
