// @webstudio-is/sdk-components-react/lib/link.ws.js
import { LinkIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { a as r } from "@webstudio-is/sdk/normalize.css";
import { props as o } from "./__generated__/link.props.js";
const n = {
  a: [
    ...r,
    {
      property: "minHeight",
      value: { type: "unit", unit: "em", value: 1 }
    },
    {
      property: "display",
      value: { type: "keyword", value: "inline-block" }
    }
  ]
}, l = {
  category: "general",
  type: "container",
  description: "Use a link to send your users to another page, section, or resource. Configure links in the Settings panel.",
  icon: e,
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  presetStyle: n,
  order: 1,
  states: [
    ...t,
    {
      selector: ":visited",
      label: "Visited"
    },
    {
      category: "component-states",
      selector: "[aria-current=page]",
      label: "Current page"
    }
  ],
  template: [
    {
      type: "instance",
      component: "Link",
      children: [
        {
          type: "text",
          value: "Link text you can edit",
          placeholder: !0
        }
      ]
    }
  ]
}, c = {
  props: {
    ...o,
    href: {
      type: "string",
      control: "url",
      required: !1
    }
  },
  initialProps: ["id", "className", "href", "target", "download"]
};
export {
  l as meta,
  c as propsMeta
};
