// @webstudio-is/sdk-components-react/lib/list.ws.js
import { ListIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { ol as o, ul as r } from "@webstudio-is/sdk/normalize.css";
import { props as p } from "./__generated__/list.props.js";
const a = {
  ol: [
    ...o,
    {
      property: "marginTop",
      value: { type: "keyword", value: "0" }
    },
    {
      property: "marginBottom",
      value: { type: "keyword", value: "10px" }
    },
    {
      property: "paddingLeft",
      value: { type: "keyword", value: "40px" }
    }
  ],
  ul: [
    ...r,
    {
      property: "marginTop",
      value: { type: "keyword", value: "0" }
    },
    {
      property: "marginBottom",
      value: { type: "keyword", value: "10px" }
    },
    {
      property: "paddingLeft",
      value: { type: "keyword", value: "40px" }
    }
  ]
}, y = {
  category: "general",
  type: "container",
  description: "Groups content, like links in a menu or steps in a recipe.",
  icon: e,
  states: t,
  presetStyle: a,
  order: 3,
  template: [
    {
      type: "instance",
      component: "List",
      children: [
        {
          type: "instance",
          component: "ListItem",
          children: [
            {
              type: "text",
              value: "List Item text you can edit",
              placeholder: !0
            }
          ]
        },
        {
          type: "instance",
          component: "ListItem",
          children: [
            {
              type: "text",
              value: "List Item text you can edit",
              placeholder: !0
            }
          ]
        },
        {
          type: "instance",
          component: "ListItem",
          children: [
            {
              type: "text",
              value: "List Item text you can edit",
              placeholder: !0
            }
          ]
        }
      ]
    }
  ]
}, c = {
  props: p,
  initialProps: ["id", "className", "ordered", "start", "reversed"]
};
export {
  y as meta,
  c as propsMeta
};
