// @webstudio-is/sdk-components-react/lib/metas.js
import { meta as o } from "./slot.ws.js";
import { meta as a } from "./fragment.ws.js";
import { meta as p } from "./html-embed.ws.js";
import { meta as s } from "./markdown-embed.ws.js";
import { meta as i } from "./content-embed.ws.js";
import { meta as d } from "./body.ws.js";
import { meta as c } from "./box.ws.js";
import { meta as B } from "./text.ws.js";
import { meta as b } from "./heading.ws.js";
import { meta as g } from "./paragraph.ws.js";
import { meta as I } from "./link.ws.js";
import { meta as V } from "./rich-text-link.ws.js";
import { meta as C } from "./span.ws.js";
import { meta as F } from "./bold.ws.js";
import { meta as R } from "./italic.ws.js";
import { meta as y } from "./superscript.ws.js";
import { meta as X } from "./subscript.ws.js";
import { meta as v } from "./button.ws.js";
import { meta as N } from "./input.ws.js";
import { meta as j, meta as z } from "./form.ws.js";
import { meta as D } from "./image.ws.js";
import { meta as J } from "./blockquote.ws.js";
import { meta as Q } from "./list.ws.js";
import { meta as W } from "./list-item.ws.js";
import { meta as Z } from "./separator.ws.js";
import { meta as $ } from "./code-text.ws.js";
import { meta as te } from "./label.ws.js";
import { meta as me } from "./textarea.ws.js";
import { meta as re } from "./radio-button.ws.js";
import { meta as xe } from "./checkbox.ws.js";
import { meta as fe } from "./vimeo.ws.js";
import { meta as ne } from "./vimeo-preview-image.ws.js";
import { meta as le } from "./vimeo-play-button.ws.js";
import { meta as ue } from "./vimeo-spinner.ws.js";
import { meta as Se } from "./xml-node.ws.js";
import { meta as Te } from "./xml-time.ws.js";
import { meta as ke } from "./time.ws.js";
import { meta as Le } from "./select.ws.js";
import { meta as he } from "./option.ws.js";
export {
  J as Blockquote,
  d as Body,
  F as Bold,
  c as Box,
  v as Button,
  xe as Checkbox,
  $ as CodeText,
  i as ContentEmbed,
  j as Form,
  a as Fragment,
  b as Heading,
  p as HtmlEmbed,
  D as Image,
  N as Input,
  R as Italic,
  te as Label,
  I as Link,
  Q as List,
  W as ListItem,
  s as MarkdownEmbed,
  he as Option,
  g as Paragraph,
  re as RadioButton,
  z as RemixForm,
  V as RichTextLink,
  Le as Select,
  Z as Separator,
  o as Slot,
  C as Span,
  X as Subscript,
  y as Superscript,
  B as Text,
  me as Textarea,
  ke as Time,
  fe as Vimeo,
  le as VimeoPlayButton,
  ne as VimeoPreviewImage,
  ue as VimeoSpinner,
  Se as XmlNode,
  Te as XmlTime
};
