// @webstudio-is/sdk-components-react/lib/props.js
import { propsMeta as p } from "./slot.ws.js";
import { propsMeta as t } from "./fragment.ws.js";
import { propsMeta as s } from "./html-embed.ws.js";
import { propsMeta as x } from "./markdown-embed.ws.js";
import { propsMeta as f } from "./body.ws.js";
import { propsMeta as n } from "./box.ws.js";
import { propsMeta as d } from "./text.ws.js";
import { propsMeta as u } from "./heading.ws.js";
import { propsMeta as S } from "./paragraph.ws.js";
import { propsMeta as b } from "./link.ws.js";
import { propsMeta as k } from "./rich-text-link.ws.js";
import { propsMeta as L } from "./span.ws.js";
import { propsMeta as h } from "./bold.ws.js";
import { propsMeta as P } from "./italic.ws.js";
import { propsMeta as w } from "./superscript.ws.js";
import { propsMeta as C } from "./subscript.ws.js";
import { propsMeta as H } from "./button.ws.js";
import { propsMeta as q } from "./input.ws.js";
import { propsMeta as N, propsMeta as O } from "./form.ws.js";
import { propsMeta as z } from "./image.ws.js";
import { propsMeta as D } from "./blockquote.ws.js";
import { propsMeta as J } from "./list.ws.js";
import { propsMeta as Q } from "./list-item.ws.js";
import { propsMeta as W } from "./separator.ws.js";
import { propsMeta as Z } from "./code-text.ws.js";
import { propsMeta as $ } from "./label.ws.js";
import { propsMeta as ro } from "./textarea.ws.js";
import { propsMeta as eo } from "./radio-button.ws.js";
import { propsMeta as ao } from "./checkbox.ws.js";
import { propsMeta as mo } from "./vimeo.ws.js";
import { propsMeta as Mo } from "./vimeo-preview-image.ws.js";
import { propsMeta as io } from "./vimeo-play-button.ws.js";
import { propsMeta as lo } from "./vimeo-spinner.ws.js";
import { propsMeta as uo } from "./xml-node.ws.js";
import { propsMeta as So } from "./xml-time.ws.js";
import { propsMeta as bo } from "./time.ws.js";
import { propsMeta as ko } from "./select.ws.js";
import { propsMeta as Lo } from "./option.ws.js";
export {
  D as Blockquote,
  f as Body,
  h as Bold,
  n as Box,
  H as Button,
  ao as Checkbox,
  Z as CodeText,
  N as Form,
  t as Fragment,
  u as Heading,
  s as HtmlEmbed,
  z as Image,
  q as Input,
  P as Italic,
  $ as Label,
  b as Link,
  J as List,
  Q as ListItem,
  x as MarkdownEmbed,
  Lo as Option,
  S as Paragraph,
  eo as RadioButton,
  O as RemixForm,
  k as RichTextLink,
  ko as Select,
  W as Separator,
  p as Slot,
  L as Span,
  C as Subscript,
  w as Superscript,
  d as Text,
  ro as Textarea,
  bo as Time,
  mo as Vimeo,
  io as VimeoPlayButton,
  Mo as VimeoPreviewImage,
  lo as VimeoSpinner,
  uo as XmlNode,
  So as XmlTime
};
