// @webstudio-is/sdk-components-react/lib/radio-button.ws.js
import { RadioCheckedIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/react-sdk";
import { radio as o } from "@webstudio-is/sdk/normalize.css";
import { props as a } from "./__generated__/radio-button.props.js";
const i = {
  input: [
    ...o,
    {
      property: "marginRight",
      value: { type: "unit", unit: "em", value: 0.5 }
    }
  ]
}, s = {
  category: "forms",
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  type: "control",
  label: "Radio",
  description: "Use within a form to allow your users to select a single option from a set of mutually exclusive choices. Group multiple radios by matching their “Name” properties.",
  icon: e,
  presetStyle: i,
  order: 5,
  states: [
    ...t,
    { selector: ":checked", label: "Checked" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ],
  template: [
    {
      type: "instance",
      component: "Label",
      label: "Radio Field",
      children: [
        { type: "instance", component: "RadioButton", props: [], children: [] },
        {
          type: "instance",
          component: "Text",
          label: "Radio Label",
          props: [{ type: "string", name: "tag", value: "span" }],
          children: [{ type: "text", value: "Radio", placeholder: !0 }]
        }
      ]
    }
  ]
}, c = {
  props: a,
  initialProps: ["id", "className", "name", "value", "required", "checked"]
};
export {
  s as meta,
  c as propsMeta
};
