// @webstudio-is/sdk-components-react/lib/separator.ws.js
import { DashIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as o } from "@webstudio-is/react-sdk";
import { hr as r } from "@webstudio-is/sdk/normalize.css";
import { props as t } from "./__generated__/separator.props.js";
const p = {
  hr: [
    ...r,
    {
      property: "height",
      value: { type: "keyword", value: "1px" }
    },
    {
      property: "backgroundColor",
      value: { type: "keyword", value: "gray" }
    },
    {
      property: "borderTopStyle",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "borderRightStyle",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "borderLeftStyle",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "borderBottomStyle",
      value: { type: "keyword", value: "none" }
    }
  ]
}, n = {
  category: "general",
  type: "embed",
  description: "Used to visually divide sections of content, helping to improve readability and organization within a webpage.",
  icon: e,
  states: o,
  presetStyle: p,
  order: 5
}, d = {
  props: t,
  initialProps: ["id", "className"]
};
export {
  n as meta,
  d as propsMeta
};
