// @webstudio-is/sdk-components-react/lib/textarea.ws.js
import { FormTextAreaIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as o } from "@webstudio-is/react-sdk";
import { textarea as t } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/textarea.props.js";
const l = {
  textarea: [
    ...t,
    // resize doesn't work well while on canvas
    { property: "resize", value: { type: "keyword", value: "none" } },
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    }
  ]
}, p = {
  category: "forms",
  type: "control",
  label: "Text Area",
  description: "A multi-line text input for collecting longer string data from your users.",
  icon: e,
  presetStyle: l,
  order: 4,
  constraints: {
    relation: "ancestor",
    component: { $nin: ["Button", "Link"] }
  },
  states: [
    ...o,
    { selector: "::placeholder", label: "Placeholder" },
    { selector: ":valid", label: "Valid" },
    { selector: ":invalid", label: "Invalid" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ]
}, c = {
  props: r,
  initialProps: [
    "id",
    "className",
    "name",
    "value",
    "placeholder",
    "required",
    "autoFocus"
  ]
};
export {
  p as meta,
  c as propsMeta
};
