// @remix-run/dev/dist/vite/vite-node.js
/**
 * @remix-run/dev v2.15.0
 *
 * Copyright (c) Remix Software Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var server = require('vite-node/server');
var client = require('vite-node/client');
var sourceMap = require('vite-node/source-map');
var importViteEsmSync = require('./import-vite-esm-sync.js');

async function createContext(viteConfig = {}) {
  await importViteEsmSync.preloadViteEsm();
  let vite = importViteEsmSync.importViteEsmSync();
  let devServer = await vite.createServer(vite.mergeConfig({
    server: {
      preTransformRequests: false,
      hmr: false
    },
    optimizeDeps: {
      noDiscovery: true
    },
    configFile: false,
    envFile: false,
    plugins: []
  }, viteConfig));
  await devServer.pluginContainer.buildStart({});
  let server$1 = new server.ViteNodeServer(devServer);
  sourceMap.installSourcemapsSupport({
    getSourceMap: source => server$1.getSourceMap(source)
  });
  let runner = new client.ViteNodeRunner({
    root: devServer.config.root,
    base: devServer.config.base,
    fetchModule(id) {
      return server$1.fetchModule(id);
    },
    resolveId(id, importer) {
      return server$1.resolveId(id, importer);
    }
  });
  return {
    devServer,
    server: server$1,
    runner
  };
}

exports.createContext = createContext;
